lua/ssr/search.lua

```diff
@@ -91,31 +91,37 @@ end
 ---@return Match[]
 function M.search(buf, node, source, ns)
   local sexpr, wildcards = build_sexpr(node, source)
-  local query = ts.parse_query(parsers.get_buf_lang(buf), sexpr)
+  local ok, query = pcall(function()
+    return ts.query.parse and ts.query.parse(parsers.get_buf_lang(buf), sexpr)
+      or ts.parse_query(parsers.get_buf_lang(buf), sexpr)
+  end)
   local matches = {}
-  local root = utils.get_root(buf)
-  for _, nodes in query:iter_matches(root, buf, 0, -1) do
-    local captures = {}
-    for var, idx in pairs(wildcards) do
-      captures[var] = ExtmarkRange:new(buf, nodes[idx], ns)
-    end
-    local match = Match:new(ExtmarkRange:new(buf, nodes[#nodes], ns), captures)
-    table.insert(matches, match)
-  end
-
-  -- Sort matches from
-  --  buffer top to bottom, to make goto next/prev match intuitive
-  --  inner to outer for recursive matches, to make replacing correct
-  table.sort(matches, function(match1, match2)
-    local start_row1, start_col1, end_row1, end_col1 = match1.range:get(buf)
-    local start_row2, start_col2, end_row2, end_col2 = match2.range:get(buf)
-    if end_row1 < start_row2 or (end_row1 == start_row2 and end_col1 <= start_col2) then
-      return true
+  if ok then
+    local root = utils.get_root(buf)
+    for _, nodes in query:iter_matches(root, buf, 0, -1) do
+      local captures = {}
+      for var, idx in pairs(wildcards) do
+        captures[var] = ExtmarkRange:new(buf, nodes[idx], ns)
+      end
+      local match = Match:new(ExtmarkRange:new(buf, nodes[#nodes], ns), captures)
+      table.insert(matches, match)
     end
-    return (start_row1 > start_row2 or (start_row1 == start_row2 and start_col1 > start_col2))
-      and (end_row1 < end_row2 or (end_row1 == end_row2 and end_col1 <= end_col2))
-  end)
 
+    -- Sort matches from
+    --  buffer top to bottom, to make goto next/prev match intuitive
+    --  inner to outer for recursive matches, to make replacing correct
+    table.sort(matches, function(match1, match2)
+      local start_row1, start_col1, end_row1, end_col1 = match1.range:get(buf)
+      local start_row2, start_col2, end_row2, end_col2 = match2.range:get(buf)
+      if end_row1 < start_row2 or (end_row1 == start_row2 and end_col1 <= start_col2) then
+        return true
+      end
+      return (start_row1 > start_row2 or (start_row1 == start_row2 and start_col1 > start_col2))
+        and (end_row1 < end_row2 or (end_row1 == end_row2 and end_col1 <= end_col2))
+    end)
+  else
+    utils.notify "Could not find treesitter parser."
+  end
```

Comment:
Parser is checked in early stage so we can assume the parser is available here, no need to pcall and notify.

Reply:
this is cleaned up :)